library/gdalframework/src/main/java/it/geosolutions/imageio/gdalframework/GDALUtilities.java

```diff
@@ -610,8 +610,58 @@ public static boolean isGDALAvailable() {
         return available;
     }
 
+    /**
+     * This will ensure that java has loaded the GDAL library (<code>gdalalljni</code> or <code>gdaljni</code>).
+     * 
+     * <ul>
+     * <li>GDAL 3+ -- GDAL.jar will automatically load the library (nothing for us to do)</li>
+     * <li>GDAL 2.3+ -- load "gdalalljni" lib</li>
+     * <li>GDAL <2.3 -- load "gdaljni" lib</li>
+     * </ul>
+     * <p>
+     * Use <code>java -XshowSettings:properties<code> to check <code>java.library.path</code>.
+     * On linux, you will probably need to set <code>LD_LIBRARY_PATH</code>.
+     */
+    private static void ensureGDALLibraryLoaded() {
+        // Called from synchronized block in loadGDAL
+        try {
+            gdal.AllRegister(); //will throw if library not already present

```

Comment:
It used to first load the library, then do an "allRegister", now it does the opposite. Does not seem a logical order, can you elaborate?